Neos.Neos/Classes/Fusion/Cache/CacheTag.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Fusion\Cache;
+
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * The cache tag value object
+ */
+#[Flow\Proxy(false)]
+class CacheTag
+{
+    protected const PATTERN = '/^[a-zA-Z0-9_%\-&]{1,250}$/';
+
+    private function __construct(
+        public readonly string $value
+    ) {
```

Comment:
shouldn't we apply the validation here to be absolutely sure?

Reply:
Hmm, we could, but then we would validate what we generate internally as the constructor is private.

Reply:
exactly, and I am sure it's fine atm, but what if we were to change the format of contentRepositoryId or anyhting else we use dynamically.

Reply:
I moved it there.